task/mgmt-gateway/src/main.rs

```diff
@@ -16,6 +16,16 @@ use task_net_api::{
 };
 use userlib::{sys_recv_closed, task_slot, TaskId, UnwrapLite};
 
+mod mgs_common;
+
+// If the build system enables multiple of the gimlet/sidecar/psc features, this
+// sequence of `cfg_attr`s will trigger an unused_attributes warning. We can
+// turn this into a hard error via this `deny`, which will catch any such build
+// system misconfiguration.
+#[deny(unused_attributes)]
+#[cfg_attr(feature = "gimlet", path = "mgs_gimlet.rs")]
```

Comment:
🤯 I never knew about this!

(and could use it to clean up BSPs in various tasks)